src/shogun/lib/parameter_observers/ObservedValueTemplated.h

```diff
@@ -39,6 +39,20 @@ namespace shogun
 			m_any_value = make_any(m_observed_value);
 		}
 
+		/**
+		 * Constructor which takes AnyParameterProperties for the observed value
+		 * @param step step
+		 * @param name the observed value's name
+		 * @param value the observed value
+		 * @param properties properties of that observed value
+		 */
+		ObservedValueTemplated(int64_t step, std::string name, T value, AnyParameterProperties properties)
+		: ObservedValue(step, name), m_observed_value(value)
+		{
+			this->watch_param("value", &m_observed_value, properties);
```

Comment:
I was thinking that the name goes in here where "value" currently is? Or am I confusing two things there?

Reply:
Here we have a variable `name` which identifies the content of the ObservedValue and the registered parameter `value` which instead store the actual value.

Reply:
ah I see of course, two layers. This will be more clear when there is an example...ill check it again then